### PR TITLE
Unruly: Add support for "siteId" bidder parameter (different casing)

### PR DESCRIPTION
**Pull request by arielunruly**
Currently, unruly.json supported only "siteid" but our ssp now supports "siteId" with a capital I.
We need to add this property to the unruly.json.

### CODE REVIEW

**Comment by hhhjort** [comment]
Won't you drop `siteId` when you unmarshal your adapter parameters and then remarshal them?

**Comment by AlexBVolcy** [comment]
I'm also wondering the same thing about `siteId` being dropped during unmarshal/marshaling

**Comment by arielunruly** [comment]
Both siteId and siteid exist for backward compatibility.

**Comment by hhhjort** [comment]
> Both siteId and siteid exist for backward compatibility.

Yes, but when you unmarshal into the struct, it will only see `siteid`, and drop `siteId`. Then when you remashal, only `siteid` can exist in the final JSON. You need to modify the struct to have both fields if you want to be able to pass both possibilities through your adapter.

**Comment by hhhjort** [comment]
You should also add a test with the alternate case to confirm the code works and generates the expected request to your servers.

**Comment by arielunruly** [comment]
@hhhjort , I have updated the struct and mocks.

File: openrtb_ext/imp_unruly.go
Comment (hhhjort) [inline openrtb_ext/imp_unruly.go:4]: You want to keep the old `siteid` entry if you want to keep supporting the old configuration. (Highly recommended for atleast a transition period). So something like:
```
type ExtImpUnruly struct {
	SiteIDOld int `json:"siteid"`
        SiteID int `json:"siteId"`
}
```
And probably have some of each type in your tests to verify that both the old and the new casings work with the current code.

Comment (AlexBVolcy) [reply openrtb_ext/imp_unruly.go:4]: Yeah agreed, having both the old configuration and the new casing is what I'd recommend.

File: static/bidder-params/unruly.json
Comment (SyntaxNode) [inline static/bidder-params/unruly.json:13]: Which casing do you prefer to use? We have a convention of ending the description of that property with `, preferred.` as a hint to publishers.

**Comment by arielunruly** [comment]
I see the tests passed except the Code scanning results Trivy, i am sorry but i haven't touched this part. what should i do ? pull again from master?

File: static/bidder-params/unruly.json
Comment (arielunruly) [reply static/bidder-params/unruly.json:13]: Do you mean to add a `,` after the end of the `description`?

**Comment by bsardo** [comment]
@arielunruly nothing is required on your part. We recently merged a change into master that will make the Trivy scan pass. We'll get another reviewer on this shortly. Thanks for your patience.

**Comment by arielunruly** [comment]
Hello @bsardo , any news with the code scanning results? :-)

File: static/bidder-params/unruly.json
Comment (AlexBVolcy) [reply static/bidder-params/unruly.json:13]: I think @SyntaxNode is wondering if you could update the description of the bidder param you prefer between `siteId` and `siteid`

So if `siteId` is the preferred casing, you can update its description to be like `"ID for publisher site, preferred"`

**Comment by AlexBVolcy** [comment]
I'll be ready to approve after you update the description of your preferred bidder param!

**Review by Sonali-More-Xandr — COMMENTED** [review]
Can you please also update the [dev-docs](https://github.com/prebid/prebid.github.io/blob/master/dev-docs/bidders/unruly.md) to reflect these changes in the parameter

**Comment by arielunruly** [comment]
> Can you please also update the [dev-docs](https://github.com/prebid/prebid.github.io/blob/master/dev-docs/bidders/unruly.md) to reflect these changes in the parameter

updated, please advise.

**Comment by AlexBVolcy** [comment]
I'm not seeing any updates to the dev-docs, or a PR for one? Could you link the changes here?

**Comment by arielunruly** [comment]
> dev-docs

yes i think i forgot to do a PR , here is the link now : 
https://github.com/prebid/prebid.github.io/pull/4381

**Comment by arielunruly** [comment]
> I'm not seeing any updates to the dev-docs, or a PR for one? Could you link the changes here?

I guess i forgot to do a PR , here is the link now : 
https://github.com/prebid/prebid.github.io/pull/4381

**Review by Sonali-More-Xandr — APPROVED** [review]

**Review by AlexBVolcy — APPROVED** [review]

**Pull request merged**
